lib/IRGen/TypeLayoutDumper.cpp

```diff
@@ -0,0 +1,212 @@
+//===--- TypeLayoutDumper.cpp ---------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines a tool for dumping layouts of fixed-size types in a simple
+// YAML format.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/AST/ASTContext.h"
+#include "swift/AST/ASTWalker.h"
+#include "swift/AST/IRGenOptions.h"
+#include "swift/AST/Types.h"
+#include "swift/SIL/SILModule.h"
+#include "swift/Subsystems.h"
+
+#include <string>
+#include <vector>
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/YAMLTraits.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "FixedTypeInfo.h"
+#include "IRGen.h"
+#include "IRGenModule.h"
+#include "GenType.h"
+#include "TypeLayoutDumper.h"
+
+using namespace swift;
+using namespace irgen;
+
+namespace {
+
+class NominalTypeWalker : public ASTWalker {
+  std::vector<NominalTypeDecl *> &Results;
```

Comment:
Instead of making a big list, maybe just have a callback? There could be a lot of these.